src/assets/scripts/client/utilities/transponderUtilities.js

```diff
@@ -0,0 +1,209 @@
+import { leftPad } from './generalUtilities';
+import { randint } from '../math/core';
+import { REGEX } from '../constants/globalConstants';
+
+/**
+ * The highest decimal value allowed for a 4-digit
+ * octal transponder code
+ *
+ * @property MAX_TRANSPONDER_CODE
+ * @type {number}
+ * @final
+ */
+const MAX_TRANSPONDER_CODE = 4095;
+
+/**
+ * List of transponder codes that are reserved
+ *
+ * This enum should be used only during the generation of
+ * `AircraftModel` objects.
+ *
+ * The codes listed should still be assignable at the
+ * controler's discretion
+ *
+ * @property TRANSPONDER_CODES
+ * @type {array<object>}
+ * @final
+ */
+const TRANSPONDER_CODES = [
+    {
+        // ICAO
+        prefix: null,
+        reserved: [
+            // Use in Mode S environment
+            '1000',
+            // When entering SSR area from non-SSR as uncontrolled IFR
+            '2000',
+            // VFR when no other code has been assigned (eg 1200 is used in the USA)
+            '7000',
+            // hijack
+            '7500',
+            // communication failure
+            '7600',
+            // emergency
+            '7700'
+        ]
+    },
+    {
+        // Europe
+        prefix: /^([elb])/,
+        reserved: [
+            '0000'
+        ]
+    },
+    {
+        // Canda
+        prefix: /^c/,
+        reserved: [
+            // VFR
+            '1200'
+        ]
+    },
+    {
+        // Belgium
+        prefix: /^eb/,
+        reserved: [
+            /^00(4[1-6]|5[0-7])$/,
+            // For testing stations
+            '7777'
+        ]
+    },
+    {
+        // Germany
+        prefix: /^e[dt]/,
+        reserved: [
+            // Parachute dropping
+            '0025',
+            // For testing stations
+            '7777'
+        ]
+    },
+    {
+        // Netherlands
+        prefix: /^eh/,
+        reserved: [
+            // For testing stations
+            '7777'
+        ]
+    },
+    {
+        // UK
+        prefix: /^eg/,
+        reserved: [
+            // Parachute drop
+            '0033'
+        ]
+    },
+    {
+        // USA
+        prefix: /^k/,
+        reserved: [
+            // Military intercept
+            '0000',
```

Comment:
FAA considers all transponder codes ending in `00` to be nondiscrete and does not assign them to specific aircraft. So `0100, 0200, 1800, 5600`, etc are all no good.

Reply:
This is checked for `isDiscreteTransponderCode`, so any non-discrete codes could be simply removed from the list of reserved items. In fact that check should be first anyway, so so it fails quickly on non-discrete codes before doing the more complex checks.

It would certainly simplify the list, but would mean they wouldn't be validated as reserved in `isReserved`. That in itself isn't a problem as the only place it's called is in `isDiscreteTransponderCode`.